### PR TITLE
mobile optimisation: reduce the size of the fragment input struct of …

### PR DESCRIPTION
**Pull request by manuele-bonanno**
### Purpose of this PR
Right now URP lighting shader (PBR, Blinn phong shading, Terrain lighting, Baked lighting) fragment input struct on average has at least 7-8 variables. In the PBR lit shader it has the following types:
float2, float2, float3, float3, float4, float3, half4, float4, float3, float4.

The aim is to reduce the size of the fragment input struct (by changing the type of variable to smaller one, for example `half`, or reducing the number of the variables in the struct itself) to free up some write bandwidth performance on vertex shader pass.

This optimisation was discussed with Apple.

---
### Testing status
local + Yamato

---
### Comments to reviewers

Vertex Shader write bandwidth before:

image

Vertex Shader write bandwidth after:

image

### CODE REVIEW

**Comment by manuele-bonanno** [comment]
The Linux Vulkan STandalone test 132 failure is a known issue on master so not related to this PR

File: com.unity.render-pipelines.universal/Shaders/Terrain/TerrainLitPasses.hlsl [hunk 1]
```diff
@@ -47,16 +47,20 @@ struct Varyings
...
-    float4 tangent                  : TEXCOORD4;    // xyz: tangent, w: viewDir.y
-    float4 bitangent                : TEXCOORD5;    // xyz: bitangent, w: viewDir.z
+    half4 normal                    : TEXCOORD3;    // xyz: normal, w: viewDir.x
+    half4 tangent                   : TEXCOORD4;    // xyz: tangent, w: viewDir.y
+    half4 bitangent                 : TEXCOORD5;    // xyz: bitangent, w: viewDir.z
 #else
-    float3 normal                   : TEXCOORD3;
-    float3 viewDir                  : TEXCOORD4;
+    half3 normal                    : TEXCOORD3;
+    half3 viewDir                   : TEXCOORD4;
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Shaders/Terrain/TerrainLitPasses.hlsl:55]: Isn't it worth it to move viewDir computation to fragment like the other shaders?

**Review by phi-lira — APPROVED** [review]
LGTM. Please add changelog entry.

**Comment by phi-lira** [comment]
ABV was green in previous changeset, just changelog was changed.

**Pull request merged**